File: android/app/src/main/java/com/dailymemory/domain/usecase/reminder/GetTodayRemindersUseCase.kt
package com.dailymemory.domain.usecase.reminder

import com.dailymemory.domain.model.Reminder
import com.dailymemory.domain.repository.ReminderRepository
import kotlinx.coroutines.flow.Flow
import java.time.LocalDateTime
import javax.inject.Inject

/**
 * Use case for getting today's reminders.
 */
class GetTodayRemindersUseCase @Inject constructor(
    private val reminderRepository: ReminderRepository
) {
    operator fun invoke(): Flow<List<Reminder>> {
        return reminderRepository.observeByDate(LocalDateTime.now())
    }
}

File: ios/DailyMemory/Domain/UseCase/Reminder/GetTodayRemindersUseCase.swift
import Foundation

/// Use case for getting today's reminders.
final class GetTodayRemindersUseCase {
    private let reminderRepository: ReminderRepository

    init(reminderRepository: ReminderRepository = ReminderRepositoryImpl()) {
        self.reminderRepository = reminderRepository
    }

    func execute() async throws -> [Reminder] {
        return try await reminderRepository.getByDate(Date())
    }
}
